h2/stream.py

```diff
@@ -668,6 +668,7 @@ def __init__(self, stream_id):
         self.state_machine = H2StreamStateMachine(stream_id)
         self.stream_id = stream_id
         self.max_outbound_frame_size = None
+        self.headers = []
```

Comment:
There's no need to store all the headers. We can avoid it simply by storing the actual value we need: specifically, the `:method` header.